packages/elasticsearch-asset-apis/src/base-api/index.ts

```diff
@@ -198,6 +194,13 @@ export class BaseReaderAPI {
         return [millisecondInterval, 'ms'];
     }
 
+    async validateSize(): Promise<void> {
```

Comment:
Since this function has a side-effect so the name should not start with `validate`, something like `setWindowSize` would more appropriate.

Reply:
hmm, though it does set the windowsize, the main purpose of this is to compare agains the size and throw if its not compatible, I will think more on the name